source/world_builder/interface.cc

```diff
@@ -0,0 +1,1082 @@
+        const bool mangle_whole_string = (s == "value");
+
+        // for all parts of the string, see
+        // if it is an allowed character or
+        // not
+        for (unsigned int i=0; i<s.size(); ++i)
+          {
+            static const std::string allowed_characters
+            ("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789");
+
+            if ((! mangle_whole_string)
+                &&
+                (allowed_characters.find (s[i]) != std::string::npos))
+              u.push_back (s[i]);
+            else
+              {
+                u.push_back ('_');
+                static const char hex[16]
+                  = { '0','1','2','3','4','5','6','7','8','9','a','b','c','d','e','f'};
+                u.push_back (hex[static_cast<unsigned char>(s[i])/16]);
+                u.push_back (hex[static_cast<unsigned char>(s[i])%16]);
+              }
+          }
+
+        return u;
+      }
+
+      std::string
+      demangle(const std::string &s)
+      {
+        std::string u;
+        u.reserve(s.size());
+
+        for (unsigned int i = 0; i < s.size(); ++i)
+          if (s[i] != '_')
+            u.push_back(s[i]);
+          else
+            {
+              Assert(i + 2 < s.size(),
+                     ExcMessage("Trying to demangle an invalid string."));
+
+              unsigned char c = 0;
+              switch (s[i + 1])
+                {
+                  case '0':
+                    c = 0 * 16;
+                    break;
+                  case '1':
+                    c = 1 * 16;
+                    break;
+                  case '2':
+                    c = 2 * 16;
+                    break;
+                  case '3':
+                    c = 3 * 16;
+                    break;
+                  case '4':
+                    c = 4 * 16;
+                    break;
+                  case '5':
+                    c = 5 * 16;
+                    break;
+                  case '6':
+                    c = 6 * 16;
+                    break;
+                  case '7':
+                    c = 7 * 16;
+                    break;
+                  case '8':
+                    c = 8 * 16;
+                    break;
+                  case '9':
+                    c = 9 * 16;
+                    break;
+                  case 'a':
+                    c = 10 * 16;
+                    break;
+                  case 'b':
+                    c = 11 * 16;
+                    break;
+                  case 'c':
+                    c = 12 * 16;
+                    break;
+                  case 'd':
+                    c = 13 * 16;
+                    break;
+                  case 'e':
+                    c = 14 * 16;
+                    break;
+                  case 'f':
+                    c = 15 * 16;
+                    break;
+                  default:
+                    Assert(false, ExcInternalError());
+                }
+              switch (s[i + 2])
```

Comment:
you can use a for loop to handle both chars with a single logic (and multiply by 16 in between)